packages/store/addon/-private/system/model/notify-changes.ts

```diff
@@ -24,14 +24,14 @@ export default function notifyChanges(
       if (meta.kind === 'belongsTo') {
         record.notifyPropertyChange(key);
       } else if (meta.kind === 'hasMany') {
-        let didRemoveUnloadedModel = false;
-        if (meta.options.async) {
-          record.notifyPropertyChange(key);
-          didRemoveUnloadedModel = internalModel.hasManyRemovalCheck(key);
-        }
-        let manyArray = internalModel._manyArrayCache[key] || internalModel._retainedManyArrayCache[key];
-        if (manyArray && !didRemoveUnloadedModel) {
-          manyArray.retrieveLatest();
+        let manyArray = internalModel._manyArrayCache[key];
+
+        if (manyArray) {
+          manyArray.notify();
+
+          if (!meta.options || meta.options.async || meta.options.async === undefined) {
```

Comment:
Can you move the comment from model notify property change to here, so that we remember why we do this